packages/context-menu/src/vaadin-menu-overlay-mixin.js

```diff
@@ -116,4 +119,48 @@ export const MenuOverlayMixin = (superClass) =>
         }
       }
     }
+
+    /**
+     * Overriding the method from OverlayFocusMixin to disable
+     * focus restoration on sub-menu overlay close. Focus should
+     * be only restored when the root menu closes.
+     *
+     * @protected
+     * @override
+     * @return {boolean}
+     */
+    _shouldRestoreFocus() {
+      if (this.parentOverlay) {
+        // Do not restore focus on sub-menu close.
+        return false;
+      }
+
+      return super._shouldRestoreFocus();
+    }
+
+    /**
+     * Overriding the method from OverlayFocusMixin to return
+     * true if the overlay contains the given node, including
+     * those within descendant menu overlays.
+     *
+     * @protected
+     * @override
+     * @param {Node} node
+     * @return {boolean}
+     */
+    _deepContains(node) {
+      // Find the closest menu overlay for the given node.
+      let overlay = getClosestElement(this.localName, node);
+      while (overlay) {
+        if (overlay === this) {
+          // The node is inside a descendant menu overlay.
+          return true;
+        }
+
+        // Traverse the overlay hierarchy to check parent overlays.
+        overlay = overlay.parentOverlay;
+      }
+
+      return false;
+    }
```

Comment:
Question: I don't know if this has been done before, but technically these two methods can't be overridden because `MenuOverlayMixin` doesn't extend `MenuOverlayMixin `. Is it fine?

Reply:
Did you mean `OverlayFocusMixin`? That's a good point. Let me add it to the extension list.